erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.js

```diff
@@ -68,52 +64,105 @@ erpnext.maintenance.MaintenanceSchedule = frappe.ui.form.Controller.extend({
 					});
 				}, __("Get Items From"));
 		} else if (this.frm.doc.docstatus === 1) {
-			this.frm.add_custom_button(__('Create Maintenance Visit'), function() {
-				frappe.model.open_mapped_doc({
-					method: "erpnext.maintenance.doctype.maintenance_schedule.maintenance_schedule.make_maintenance_visit",
-					source_name: me.frm.doc.name,
-					frm: me.frm
+			var s = me.frm.doc.schedules;
+			let flag = 0;
+			for (let i in s) {
+				if (s[i].completion_status == "Pending") {
+					flag = 1;
+				}
+			}
+			if (flag) {
+				this.frm.add_custom_button(__('Create Maintenance Visit'), function () {
+					let options = "";
+					
+					me.frm.call('get_pending_data',{data_type:"items"}).then(r =>{
+						options = r.message
+						
+					var schedule_id = "";
+					var d = new frappe.ui.Dialog({
+						title: __("Enter Visit Details"),
+						fields: [{
+							fieldtype: "Select",
+							fieldname: "item_name",
+							label: __("Item Name"),
+							options: options,
+							reqd: 1,
+							onchange: function () {
+								let field = d.get_field("scheduled_date");
+								me.frm.call('get_pending_data',{item_name:this.value,data_type:"date"}).then(r =>{
+									field.df.options = r.message;
+									field.refresh();
+								})
+							}
+						},
+						{
+							label: __('Scheduled Date'),
+							fieldname: 'scheduled_date',
+							fieldtype: 'Select',
+							options: "",
+							reqd: 1,
+							onchange: function () {
+								let field = d.get_field('item_name');
+								me.frm.call('get_pending_data',{item_name:field.value,s_date:this.value,data_type:"id"}).then(r =>{
```

Comment:
Readability is important, make sure the method, args are all readable and one after the other. (Mind the indentation, github doesnt help so its messy)
```suggestion
								me.frm.call(
								    'get_pending_data',
								    {
								        item_name: field.value,
								        s_date: this.value,
								        data_type: "id"
								    }).then(r => {
```